kafka_listener.py

```diff
@@ -0,0 +1,63 @@
+import json
+
+from kafka import KafkaConsumer
+
+from historical_system_profiles import config, listener_logging
+from historical_system_profiles.app import create_app
+from historical_system_profiles import db_interface
+
+
+def main():
+    logger = listener_logging.initialize_logging()
+    logger.error("starting listener in %s mode" % config.listener_mode)
+
+    consumer = init_consumer(config.listener_mode)
+
+    app = create_app()
+    # grab the context from the flask app instead of the connexion app
+    with app.app.app_context():
+        while True:
+            for data in consumer:
+                logger.info("consuming message")
+                try:
+                    if config.listener_mode == "ARCHIVER":
+                        host = data.value["host"]
+                        db_interface.create_profile(
+                            inventory_id=host["id"],
+                            profile=host["system_profile"],
+                            account_number=host["account"],
+                        )
+                        logger.info(
+                            "wrote inventory_id %s's profile to historical database"
+                            % host["id"]
+                        )
+                    elif config.listener_mode == "DELETER":
+                        if data.value["type"] == "delete":
+                            inventory_id = data.value["id"]
+                            db_interface.delete_hsps_by_inventory_id(inventory_id)
+                            logger.info(
+                                "deleted profiles for inventory_id %s" % inventory_id
+                            )
```

Comment:
It would be nice if the code for archiver and deleter were broken out into their own functions to encapsulate the logic. This big try-if-else block gets hard to parse.